extended/src/main/java/apoc/ml/OpenAI.java

```diff
@@ -111,13 +113,26 @@ public Stream<EmbeddingResult> getEmbedding(@Name("texts") List<String> texts, @
       "model": "text-embedding-ada-002",
       "usage": { "prompt_tokens": 8, "total_tokens": 8 } }
     */
-        Stream<Object> resultStream = executeRequest(apiKey, configuration, "embeddings", "text-embedding-ada-002", "input", texts, "$.data", apocConfig, urlAccessChecker);
-        return resultStream
+        Map<Boolean, List<String>> collect = texts.stream()
+                .collect(Collectors.groupingBy(Objects::nonNull));
+
+        List<String> nonNullTexts = collect.get(true);
```

Comment:
qui non servirebbe vedere se la collezione è nulla di modo da poter lanciare un eccezione prima di fare la richiesta?

Reply:
Sì in effetti è meglio, ho aggiunto il check con l'eccezione sopra.
Questa parte l'ho lasciata cosi perché è richiesto dalla issue:
```
But we should probably just filter those out in the embeddings call and don't forward them and return the result row with a null value for embedding.
```